File: ParrotSpam.fs
open System

let MAX_CHARS = 4000

let cycle(s : seq<'a>) : seq<'a> = seq { while true do yield! s }

[<EntryPoint>]
let main(args : string[]) : int =
    let parrots = Array.filter (not << Seq.isEmpty) args
    if Seq.isEmpty parrots then
        Console.WriteLine "Nothing to repeat"
        exit 1

    let mutable used = 0
    let consuming(p : string) : bool =
        used <- used + p.Length
        used < MAX_CHARS

    parrots |> cycle |> Seq.takeWhile consuming |> Seq.iter Console.Write
    if not Console.IsOutputRedirected then
        Console.WriteLine ()

    0

File: parrot_spam.rs
use std::env;
use std::process;

const MAX_CHARS: usize = 4000;

fn main() {
    let mut used = 0usize;

    let args: Vec<String> = env::args()
        .skip(1)
        .filter(|x| !x.is_empty())
        .collect();

    if args.is_empty() {
        eprintln!("Nothing to repeat");
        process::exit(1);
    }

    for arg in args.iter().cycle().take_while(|&x| {
        used += x.len();
        used <= MAX_CHARS
    }) {
        print!("{}", arg);
    }
}
